packages/editor/src/browser/editor-widget.ts

```diff
@@ -6,19 +6,23 @@
  */
 
 import { SelectionService } from '@theia/core/lib/common';
-import { Widget, BaseWidget, Message } from '@theia/core/lib/browser';
-import { TextEditor } from "./editor";
+import { Widget, BaseWidget, Message, SaveableSource } from '@theia/core/lib/browser';
+import { TextEditor, TextEditorDocument } from "./editor";
 
-export class EditorWidget extends BaseWidget {
+export class EditorWidget extends BaseWidget implements SaveableSource {
 
     constructor(
         readonly editor: TextEditor,
-        readonly selectionService: SelectionService
+        protected readonly selectionService: SelectionService
     ) {
         super(editor);
         this.toDispose.push(this.editor);
     }
 
+    get saveable(): TextEditorDocument {
```

Comment:
should the return type not only be `Saveable`. We already expose the document through `editor.document`.